### PR TITLE
[native] Add MacOS build to CircleCI

### PR DESCRIPTION
**Pull request by v-jizhang**
* Add MacOS build to CircleCI for native-specific PRs and fix "Illegal instruction: 4" error we had previously.

Test plan - Make sure CircleCi jobs pass.
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ workflows:
   dist-compile:
     jobs:
       - linux-build
+      - native-specific-macos
```
Comment (mshang816) [inline .circleci/continue_config.yml:24]: Should we move this line after "conditionals" below ?

File: .circleci/continue_config.yml [hunk 2]
```diff
@@ -69,7 +78,7 @@ jobs:
           name: 'Maven Install'
           command: |
             export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-            ./mvnw clean install ${MAVEN_FAST_INSTALL} -pl 'presto-native-execution' -am
+            for i in $(seq 1 3); do ./mvnw clean install ${MAVEN_FAST_INSTALL} -pl 'presto-native-execution' -am && s=0 && break || s=$? && sleep 10; done; (exit $s)
```
Comment (mshang816) [inline .circleci/continue_config.yml:81]: Can we move this one to a separate PR ?

File: .circleci/continue_config.yml [hunk 3]
```diff
@@ -42,6 +43,14 @@ executors:
   check:
     docker:
       - image: prestocpp/velox-check:mikesh-20210609
+  macos-intel:
+    macos:
+      xcode: "14.3.0"
+    resource_class: macos.x86.medium.gen2
+  macos-m1:
```
Comment (mshang816) [inline .circleci/continue_config.yml:50]: it looks like we are not using m1 so we'd probably better remove macos-m1 here

File: .circleci/continue_config.yml (hunk 1 above)
Comment (v-jizhang) [reply .circleci/continue_config.yml:24]: Yes, we don't have any tests for macOS build. It should be conditional.

File: .circleci/continue_config.yml (hunk 2 above)
Comment (v-jizhang) [reply .circleci/continue_config.yml:81]: Sure

File: .circleci/continue_config.yml [hunk 4]
```diff
@@ -131,3 +140,31 @@ jobs:
...
+    executor: macos-intel
+    steps:
+      - checkout
+      - run:
+          name: "Update submodules"
+          command: |
+            cd presto-native-execution
+            make submodules
+      - run:
+          name: "Setup MacOS"
```
[the remark this answers is not in the document]
Comment (v-jizhang) [reply .circleci/continue_config.yml:154]: Ok

File: .circleci/continue_config.yml (hunk 3 above)
Comment (v-jizhang) [reply .circleci/continue_config.yml:50]: Ok.

**Review by mshang816 — APPROVED** [review]
LGTM

**Review by ajaygeorge — APPROVED** [review]
Approving since Michael took a look.

**Comment by v-jizhang** [comment]
@prestobot kick off tests
